packages/plugin-core/src/commands/InstrumentedWrapperCommand.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from "vscode";
+import { DENDRON_COMMANDS } from "../constants";
+import { AnalyticsUtils } from "../utils/analytics";
+import { BasicCommand } from "./base";
+
+export type InstrumentedWrapperCommandArgs = {
+  /**
+   * The underlying command to be wrapped
+   */
+  command: vscode.Command;
+
+  /**
+   * telemetry event name
+   */
+  event: string;
+
+  /**
+   * custom props to attach to the telemetry event
+   */
+  customProps?: any;
+};
+
+/**
+ * This command is a simple wrapper around commands, but this wrapper also fires
+ * a telemetry event when it is invoked. This is intended to be used as a
+ * wrapper around built-in VSCode commands that need to be invoked via a command
+ * URI, such as within webviews or in TreeView items, but where we still want a
+ * telemetry event to get reported.
+ */
+export class InstrumentedWrapperCommand extends BasicCommand<
```

Comment:
can we do an async and add it to the instrumentation cookbook after the pr has been merged?

Reply:
sure